slsDetectorServers/gotthard2DetectorServer/slsDetectorFunctionList.c

```diff
@@ -748,12 +758,39 @@ int64_t getNumFrames() {
 void setNumTriggers(int64_t val) {
     if (val > 0) {
 		FILE_LOG(logINFO, ("Setting number of triggers %lld\n", (long long int)val));
-        set64BitReg(val, SET_CYCLES_LSB_REG, SET_CYCLES_MSB_REG);
+		numTriggers = val;
+		if (burstMode != BURST_OFF && getTiming() == AUTO_TIMING) {
+			FILE_LOG(logINFO, ("\tBurst and Auto mode: not writing #triggers to register\n"));
+		} else {
+        	set64BitReg(val, SET_CYCLES_LSB_REG, SET_CYCLES_MSB_REG);
+		}
     } 
 }
 
 int64_t getNumTriggers() {
-    return get64BitReg(SET_CYCLES_LSB_REG, SET_CYCLES_MSB_REG);
+	if (burstMode != BURST_OFF && getTiming() == AUTO_TIMING) {
+		return numTriggers;
+	}
+	return get64BitReg(SET_CYCLES_LSB_REG, SET_CYCLES_MSB_REG);
+}
+
+void setNumBursts(int64_t val) {
+    if (val > 0) {
+		FILE_LOG(logINFO, ("Setting number of bursts %lld\n", (long long int)val));
+		numBursts = val;
+		if (burstMode != BURST_OFF && getTiming() == AUTO_TIMING) {
+			set64BitReg(val, SET_CYCLES_LSB_REG, SET_CYCLES_MSB_REG);
+		} else {
+        	FILE_LOG(logINFO, ("\tNot (Burst and Auto mode): not writing #bursts to register\n"));
+		}
+    } 
+}
+
+int64_t getNumBursts() {
+	if (burstMode != BURST_OFF && getTiming() == AUTO_TIMING) {
+		return get64BitReg(SET_CYCLES_LSB_REG, SET_CYCLES_MSB_REG);
+	}
+	return numBursts;
 }
 
 int setExpTime(int64_t val) {
```

Comment:
Not displayed here but looks like setExptimeBurst and setExptimeCont calls the same function. Maybe I misunderstood the logic but seems confusing.

Reply:
Yes. This is for now. Soon, the plan is to have different registers.